### ISSUE TITLE
arrow-body-style as-needed to to exclude multi line with comments

### ISSUE THREAD

**Issue by gregid** [eslint/eslint#13438]
**What rule do you want to change?**

`arrow-body-style`

**Does this change cause the rule to produce more or fewer warnings?**

fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option, e.g. `ignoreComments: false` to work in conjunction with `as-needed`

**Please provide some example code that this change will affect:**

currently:
```js
let foo = () => { return something(); };
let bar = () => { 
    // here is some commented out code
    // operation1();
    // operation2();
    return something(); 
};
```
Will be expected/fixed to:
```js
let foo = () => something();
let bar = () => 
    // here is some commented out code
    // operation1();
    // operation2();
    something();
```

**What does the rule currently do for this code?**

Currently `as-needed` will move the code outside of block even if there are multiple lines of commented code.

**What will the rule do after it's changed?**

With `ignoreComments: false` it will skip recommending to move the code outside of block if there are comments in the block.
In the example above `foo` will be fixed as before, while `bar` will be left unchanged (in a block)

**Are you willing to submit a pull request to implement this change?**

Not for a while.

**Comment by anikethsaha** [comment]
THanks for the issue, 

I think this will fall under the [new policy](https://eslint.org/blog/2020/05/changes-to-rules-policies#what-are-the-changes). 

if not, this seems like a good enhancement.

**Comment by gregid** [comment]
In general I believe that current rule is error prone and not very "linty", the suggestion with another option is in order to avoid breaking changes in behavior. But I would be simply happy if eslint would stop correcting `bar` at all.

**Comment by kaicataldo** [comment]
I don't believe this should be considered a bug, and would therefore fall under the rule policy @anikethsaha mentions above. Given the intention of the rule (which is consistent use of implicit return in arrow functions), I think the above autofix is correct. ESLint has no way of understanding the intention of the comment, and ignoring comments could lead to false negatives such as:

```
let bar = () => { 
    // Here is a description with code samples that were never intended to be run.
    // operation1();
    // operation2();
    return something(); 
};

let bar = () => { 
    // Here's a comment about the something()'s side effects.
    return something(); 
};
```
should be fixed to:

```
let bar = () =>
    // Here is a description with code samples that were never intended to be run.
    // operation1();
    // operation2();
    something(); 

let bar = () =>
    // Here's a comment about the something()'s side effects.
    something();
```

**Comment by gregid** [comment]
My argument is that even the second case (1 line of comment) should stay in block or to put it in other words the (no return) arrow function should NOT be separated by a comment (no matter the context/intent of comment).
Such a comment should be moved above the arrow function instead with no harm to the comment's meaning.

This would have added benefit of being able to force eslint to leave the block to the code you want in the block by simply adding a comment line and removing/moving the comment when you are happy with it. This is helpful in development when you expect there may be more lines and don't want to constantly envelop the existing code in block when adding new line and remember about adding return.

But since I understand there may be disagreement here my original proposal is with new option.

**Comment by kaicataldo** [comment]
> My argument is that even the second case (1 line of comment) should stay in block or to put it in other words the (no return) arrow function should NOT be separated by a comment (no matter the context/intent of comment).
Such a comment should be moved above the arrow function instead with no harm to the comment's meaning.

This comes down to personal preference, and I don't think is a compelling reason for inclusion given our new policy to freeze stylistic rules (please see https://eslint.org/blog/2020/05/changes-to-rules-policies for more details).

That being said, I understand the difficulty of the specific situation you're describing (I'm assuming you're using a text editor integration to format the code on save). One example of a case where I think this new behavior would lead complaints about it working the opposite way is the following:

```js
let foo = () => { /* inline comment */ return something(); };
```

In this proposal, this would not generate a warning.
